{{cookiecutter.project_slug}}/Makefile

```diff
@@ -1,111 +1,136 @@
-##### PATHS #####
-
 VERSION=v1.4
 BASE_ENV_VERSION=v1.4
 PROJECT_ID=placeholder
 
-DATA_DIR?=data
-CONFIG_DIR?=config
-CODE_DIR?={{cookiecutter.code_directory}}
-NOTEBOOKS_DIR?=notebooks
-RESULTS_DIR?=results
+##### CONSTANTS #####
 
-PROJECT_PATH_STORAGE?=storage:{{cookiecutter.project_slug}}
+DATA_DIR=data
+CONFIG_DIR=config
+CODE_DIR={{cookiecutter.code_directory}}
+NOTEBOOKS_DIR=notebooks
+RESULTS_DIR=results
 
-PROJECT_PATH_ENV?=/project
+PROJECT_PATH_STORAGE=storage:{{cookiecutter.project_slug}}
+PROJECT_PATH_ENV=/project
 
-##### JOB NAMES #####
-PROJECT?={{cookiecutter.project_slug}}
-SETUP_JOB?=setup-$(PROJECT)
-TRAIN_JOB?=train-$(PROJECT)
-DEVELOP_JOB?=develop-$(PROJECT)
-JUPYTER_JOB?=jupyter-$(PROJECT)
-TENSORBOARD_JOB?=tensorboard-$(PROJECT)
-FILEBROWSER_JOB?=filebrowser-$(PROJECT)
+PROJECT={{cookiecutter.project_slug}}
+SETUP_JOB=setup-$(PROJECT)
+TRAIN_JOB=train-$(PROJECT)
+DEVELOP_JOB=develop-$(PROJECT)
+JUPYTER_JOB=jupyter-$(PROJECT)
+TENSORBOARD_JOB=tensorboard-$(PROJECT)
+FILEBROWSER_JOB=filebrowser-$(PROJECT)
 
-##### ENVIRONMENTS #####
+BASE_ENV_NAME=neuromation/base:$(BASE_ENV_VERSION)
+CUSTOM_ENV_NAME=image:neuromation-$(PROJECT):$(VERSION)
 
-BASE_ENV_NAME?=neuromation/base:$(BASE_ENV_VERSION)
-CUSTOM_ENV_NAME?=image:neuromation-$(PROJECT):$(VERSION)
 
-##### VARIABLES YOU MAY WANT TO MODIFY #####
+##### VARIABLES #####
 
-# Jupyter mode. Available options: notebook (to run Jupyter Notebook), lab (to run JupyterLab).
-JUPYTER_MODE?=notebook
+# To overload a variable use either way:
+# - change its default value in Makefile,
+# - export variable: `export VAR=value`,
+# - or change its value for a single run only: `make <target> VAR=value`.
 
-# Location of your dataset on the platform storage. Example:
-# DATA_DIR_STORAGE?=storage:datasets/cifar10
+# Allows to set the `neuro` executable:
+#   make setup NEURO=/usr/bin/neuro
+#   make setup NEURO="neuro --verbose --show-traceback"
+NEURO?=neuro
+
+# Location of your dataset on the platform storage:
+#   make setup DATA_DIR_STORAGE=storage:datasets/cifar10
 DATA_DIR_STORAGE?=$(PROJECT_PATH_STORAGE)/$(DATA_DIR)
 
-# The type of the training machine (run `neuro config show` to see the list of available types).
+# The type of the training machine (run `neuro config show`
+# to see the list of available types):
+#   make jupyter PRESET=cpu-small
 PRESET?=gpu-small
 
+# Extra options for `neuro run` targets:
+#   make train RUN_EXTRA="--env MYVAR=value"
+RUN_EXTRA?=
```

Comment:
Can you please provide more examples of when this feature may be useful?

I see an example with TensorFlow logs in the issue, and it seems to be a rare case (I may be wrong here, of course).

Reply:
this is an argument to run a job with some custom extra arguments without modifying Makefile:
`make train RUN_EXTRA="--env MYVAR=value --volume storage:mine:/mine:ro"`